hal/ia32/pmap.c

```diff
@@ -278,23 +277,23 @@ int pmap_getPage(page_t *page, addr_t *addr)
 	}*/
 
 	hal_spinlockClear(&pmap_common.lock, &sc);
+#endif
 
-	if ((page->addr >= (u32)syspage - VADDR_KERNEL) && (page->addr < (u32)syspage - VADDR_KERNEL + SIZE_PAGE)) {
+	/* Calculate syspage size */
+	sz = *(ptr_t *)((ptr_t)syspage + sizeof(hal_syspage_t));
```

Comment:
Why not `syspage->size`?

Reply:
`syspage` is a type of `hal_syspage_t *`. `size` is a part of generic syspage structure `syspage_t.`

Reply:
It would be better to handle main `syspage` symbol here. You include syspage.h anyway, so why not to access generic syspage by `syspage->field` and `hal_syspage_t` by `syspage->hs->field`?

Reply:
The main assumption was that hal has an access only to `hal_syspage_t` part of the syspage structure. 
`pmap_getPage` uses fields from generic part of syspage but it can be moved outside of hal, on each of the platform with MMU code looks similar.

Your proposition will solve the problem at the moment but I think that we should follow the rule that hal gets from the outside world only the minimum.